build.sh

```diff
@@ -15,26 +15,61 @@ echo "Installing dotnet CLI"
 mkdir -p cli
 curl -o cli/dotnet-install.sh -L https://dot.net/v1/dotnet-install.sh
 
+if [ $? -ne 0 ]; then
+	echo "Could not download 'dotnet-install.sh' script. Please check your network and try again!"
+	exit 1
+fi
+
 # Run install.sh for cli
 chmod +x cli/dotnet-install.sh
 
-# v1 needed for some test and bootstrapping testing version
-cli/dotnet-install.sh -i cli -c 1.0
-
-DOTNET="$(pwd)/cli/dotnet"
-
-echo "$DOTNET msbuild build/config.props /v:m /nologo /t:GetCliBranchForTesting"
+# Get recommended version for bootstrapping testing version
+cli/dotnet-install.sh -i cli -c 2.2
 
-# run it twice so dotnet cli can expand and decompress without affecting the result of the target
-$DOTNET msbuild build/config.props /v:m /nologo /t:GetCliBranchForTesting
-DOTNET_BRANCH="$($DOTNET msbuild build/config.props /v:m /nologo /t:GetCliBranchForTesting)"
+if [ $? -ne 0 ]; then
+	echo ".NET CLI Install failed!!"
+	exit 1
+fi
 
-echo $DOTNET_BRANCH
-cli/dotnet-install.sh -i cli -c $DOTNET_BRANCH
+DOTNET="$(pwd)/cli/dotnet"
 
-# Display current version
+# Let the dotnet cli expand and decompress first if it's a first-run
 $DOTNET --version
 
+# Get CLI Branches for testing
+echo "dotnet msbuild build/config.props /v:m /nologo /t:GetCliBranchForTesting"
+
+IFS=$'\n'
+CMD_OUT_LINES=(`$DOTNET msbuild build/config.props /v:m /nologo /t:GetCliBranchForTesting`)
+# Take the line just before the last empty line and remove all the spaces
+DOTNET_BRANCHES=${CMD_OUT_LINES[-1]//[[:space:]]}
+unset IFS
+
+IFS=$';'
+for DOTNET_BRANCH in ${DOTNET_BRANCHES[@]}
+do
+	echo $DOTNET_BRANCH
+
+	IFS=$':'
+	ChannelAndVersion=($DOTNET_BRANCH)
+	Channel=${ChannelAndVersion[0]}
+	if [ ${#ChannelAndVersion[@]} -eq 1 ]
+	then
+		Version="latest"
+	else
+		Version=${ChannelAndVersion[1]}
+	fi
+	unset IFS
+
+	echo "Channel is: $Channel"
+	echo "Version is: $Version"
+	cli/dotnet-install.sh -i cli -c $Channel -v $Version -nopath
+
+	# Display current version
+	$DOTNET --version
+	dotnet --info
```

Comment:
This should be `$DOTNET --info`, so we see the information from the .NET SDK installed in the `cli/` directory. With this change, it's only showing system installed SDKs and runtimes, so the logs don't help us confirm that the .NET SDK was correctly installed into the `cli/` folder.

Personally, I think we should also set `DOTNET_MULTILEVEL_LOOKUP=0`, and maybe `DOTNET_HOME`, or whatever other environment variables are important to the dotnet cli, so it doesn't also list the system installed SDK and runtimes. But we don't do that anywhere else.